test/run_pass/test_iotjs_promise_then_bind.js

```diff
@@ -0,0 +1,13 @@
+var assert = require('assert');
+
+class TestObj {
+  constructor () {}
+  then() {
+    assert (this === obj)
+  }
+}
+
+var obj = new TestObj();
+var promise = new Promise(() => {
```

Comment:
`new Promise` should be prevented from using apart from `resolver`.

Suggestion:
```js
Promise.resolve().then(() => obj)
```

Reply:
OK, updated.